Detectors/TPC/calibration/SpacePoints/src/ResidualAggregator.cxx

```diff
@@ -195,15 +218,41 @@ void ResidualAggregator::finalizeSlot(Slot& slot)
   cont->treeOutResiduals.reset();
   cont->treeOutStats->Write();
   cont->treeOutStats.reset();
+  cont->treeOutRecords->Write();
+  cont->treeOutRecords.reset();
   cont->fileOut->Close();
   cont->fileOut.reset();
+  std::filesystem::rename(o2::utils::Str::concat_string(mOutputDir, cont->fileName, ".part"), mOutputDir + cont->fileName);
+  if (mStoreMetaData) {
+    mFileMetaData = std::make_unique<o2::dataformats::FileMetaData>();
+    mFileMetaData->fillFileData(mOutputDir + cont->fileName);
+    // mFileMetaData->run = getTFrunNumber();
```

Comment:
Why did you comment setting the run number?

Reply:
I actually forgot to fix this, that is why. I am taking the run number now from the TimeSlot which in turn gets it through the TFIDInfo. Thanks for spotting!